### DISCUSSION TITLE
[SOLVED] [STM32H503] SPI1 does not send anything

### DISCUSSION THREAD

**Discussion by smalcom** [stm32duino/Arduino_Core_STM32#2595] [Libraries]
Good day, everyone.

I'm trying to launch SPI1 under Arduino IDE. The transmission didn't work and I switched to direct control of the registers.

The problem is solved. By default, stm32duino sets the system clocking from PLL1P. And SPI1 cannot use this signal as a source. I switched SPI1 to clocking from PER_CK and everything worked.

Copy of the [stthread](https://community.st.com/t5/stm32-mcus-products/solved-stm32h503-spi1-does-not-send-anything/m-p/751618)

**Comment by fpistm** [comment]
Hi @smalcom 
in fact there is 2 issues, first is the HSE should not be used as stated in the errata sheet.
Second, the SPI input freq is simply to high.

I've created an issue from this discussion: #2598 

Note:
by default SPI1 is feed by PLL1Q not PLL1P.

**Comment by smalcom** [comment]
Cool, superfast fix.
Thank you.
